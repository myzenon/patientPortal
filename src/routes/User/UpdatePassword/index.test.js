import React from 'react'
import { UpdatePassword } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

const mockToken = 'eyJlbWFpbCI6ICJzaS56ZW5vbkBnbWFpbC5jb20iLCAicGF0aWVudF9pZCI6ICI3YzczMzBjMC1hMzIxLTRjYmEtODE1My1jMTVlOWU4N2ZiZGYiLCAib3JnX2lkIjogImY5MWIxOWQyLTU1ZTUtNDcyZC04ZTllLWFmMmIzZjcyOTIxZCJ9'
const mockCode = '480348'
const mockParams = `?token=${mockToken}&code=${mockCode}`
const mockSecurityPolicy = {
    "sessionTimeout": "15",
    "passwordExpiration": "1825",
    "passwordPattern": /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\\$%\\^&\\*\\=])(?=.{8,})/,
    "passwordMsg": "Password must be at least 8 characters long and contain at least one character, one number and one special character including !@#$%*="
}

test.skip('renders correctly | on page loading | call /users/organization-policy?token=xxxx&code=xxxx', (done) => {
    const props = {
        callAPIGateway: (name, get) => {
            expect(name).toBe('usersOrganizationPolicyGet')
            expect(get).toEqual({
               token: mockToken
            })
            return Promise.resolve()
        },
        router: {
            location: {
                search: mockParams
            }
        },
        page: {
            apiData: {
                ...mockSecurityPolicy
            }
        }
    }
    const wrapper = mount (
        <UpdatePassword {...props} />
    )
    setTimeout(() => {
        expect(toJson(wrapper.render())).toMatchSnapshot()
        done()
    }, 0)
})

test('no token and code parameter | redirect to login page.', done => {
    const props = {
        router: {
            location: {
                search: ''
            }
        },
        push: (url) => {
            expect(url).toBe('/login')
            done()
        }
    }
    mount (
        <UpdatePassword {...props} />
    )
})

test.skip('toggle switch showing password | show password between as clear text and as *******', (done) => {
    const props = {
        callAPIGateway: (name, get) => {
            expect(name).toBe('usersOrganizationPolicyGet')
            expect(get).toEqual({
               token: mockToken
            })
            return Promise.resolve()
        },
        router: {
            location: {
                search: mockParams
            }
        },
        page: {
            apiData: {
                ...mockSecurityPolicy
            }
        }
    }
    const wrapper = mount (
        <UpdatePassword {...props} />
    )
    setTimeout(() => {
        wrapper.update()
        wrapper.find('span[className="switch switchOff"]').simulate('click')
        expect(wrapper.find('input[name="password"]').props().type).toBe('text')
        wrapper.find('span[className="switch switchOn"]').simulate('click')
        expect(wrapper.find('input[name="password"]').props().type).toBe('password')
        done()
    }, 0)
})

test.skip('on update password | update password api called', done => {
    const mockPassword = 'A12345678*a'
    const props = {
        callAPIGateway: (name, get) => {
            expect(name).toBe('usersOrganizationPolicyGet')
            expect(get).toEqual({
               token: mockToken
            })
            return Promise.resolve()
        },
        router: {
            location: {
                search: mockParams
            }
        },
        page: {
            apiData: {
                ...mockSecurityPolicy
            }
        },
        updatePassword: (creds) => {
            expect(creds).toEqual({
                creds: {
                    token: mockToken,
                    code: mockCode,
                    password: mockPassword
                }
            })
            done()
            return Promise.resolve()
        }
    }
    const wrapper = mount (
        <UpdatePassword {...props} />
    )
    setTimeout(() => {
        wrapper.update()
        wrapper.find('input[name="password"]').simulate('change', {
            target: {
                name: 'password',
                value: mockPassword
            }
        })
        wrapper.find('button[type="submit"]').simulate('submit')
    }, 0)
})
