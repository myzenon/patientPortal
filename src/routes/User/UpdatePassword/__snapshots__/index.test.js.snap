// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly | on page loading | call /users/organization-policy?token=xxxx&code=xxxx 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <p>
      Let's get you a new password.
    </p>
    <div
      class="formWrapper"
    >
      <form
        class=""
      >
        <div>
          <input
            class="inputText"
            name="password"
            placeholder="enter new password"
            type="password"
            value=""
          />
        </div>
        <div>
          <span
            class="switch switchOff"
          />
              Show Password
        </div>
        <div
          class="passwordRule"
        >
          For security purposes, Password must be at least 8 characters long and contain at least one character, one number and one special character including !@#$%*=
        </div>
        <div>
          <button
            class="button btn btn-secondary"
            type="submit"
          >
            Change Password and Sign In
          </button>
        </div>
      </form>
    </div>
  </div>
  <div
    class="footerWrapper container"
  >
    <p>
      <strong>
        Hint:
      </strong>
       From now on, you will log in with the email shown on this page and the password you set in the above box
    </p>
  </div>
</div>
`;
